pkg/scheduler/api/devices/nvidia/vgpu/utils.go

```diff
@@ -326,14 +327,15 @@ func getGPUDeviceSnapShot(snap *GPUDevices) *GPUDevices {
 }
 
 // checkNodeGPUSharingPredicate checks if a pod with gpu requirement can be scheduled on a node.
-func checkNodeGPUSharingPredicate(pod *v1.Pod, gssnap *GPUDevices, replicate bool) (bool, []ContainerDevices, error) {
+func checkNodeGPUSharingPredicate(pod *v1.Pod, gssnap *GPUDevices, replicate bool) (bool, []ContainerDevices, float64, error) {
```

Comment:
predict and score are both in one function, but funcname is predict, a little wired here. Can we seperate them?

Reply:
i thought about that, but either "predicate" or "score" needs to iterate all available devices on this node, which is a rather slow move. so i calculate and cache the score in "predicate", and return the corresponding score directly in "score" in order to improve performance

Reply:
If so, we'd better rename this function name.

Reply:
ok, i 'll rename it to checkNodeGPUSharingPredicateWithScore(...)